rfc/produce-message-size.md

```diff
@@ -0,0 +1,99 @@
+# Enhancing Fluvio to Message Sizes
+
+This RFC proposes modifications to Fluvio's handling of message sizes. 
+
+## Proposed Enhancements
+
+1. Handling Larger Messages than Batch Size
+
+If a single record exceeds the defined `batch_size`, Fluvio will process the record as a standalone request, ensuring that larger messages are not discarded or delayed. If the record does not exceed the `batch_size`, Fluvio will process the record as part of an already existing batch or create a new one if the batch is full.
+
+2. Handling Larger Messages than the Max Request Size
+
+Fluvio will have a new configuration parameter, max_request_size, that will define the maximum size of a request that can be sent by the producer. This configuration will make Fluvio display errors when a message exceeds the defined `max_request_size`, even if it's a message with only one record or a batch of records.
+
+3. Compression Behavior
+
+Fluvio will ensure that configuration limits that use size constraints, such as `batch_size` and `max_request_size` will only use the uncompressed message size.
```

Comment:
That's the problem we need to solve. The algorithm needs to compute the result `after` compression, as that's what's carried in the payload.

Reply:
As batch is a concept before the message, I really think that we should not use or try to guess the compression size for that. 

Right now, Fluvio is just "tring to guess" the compression, multiplying every uncompressed size to half (0.5) when it's a compressed batch.

About `max_request_size` I don't see problem to use compression size for that, but Kafka does not use the compression size for `max_request_size`. So if we want that, it's better use another name to not generate confusion for the clients.